src/main/java/tc/oc/pgm/rotation/RotationManager.java

```diff
@@ -0,0 +1,218 @@
+package tc.oc.pgm.rotation;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.commons.io.FileUtils;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.configuration.file.FileConfiguration;
+import org.bukkit.configuration.file.YamlConfiguration;
+import org.bukkit.entity.Player;
+import tc.oc.pgm.api.PGM;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.match.MatchManager;
+import tc.oc.pgm.map.PGMMap;
+
+public class RotationManager implements PGMMapOrderProvider {
+
+  private MatchManager matchManager;
+
+  private File rotationsFile;
+  private List<Rotation> rotations = new ArrayList<>();
+  private Rotation activeRotation;
+  private boolean isEvaluatingPlayerCount = true;
+  /** When a {@link PGMMap} is manually set next, it overrides the rotation order * */
+  private PGMMap overriderMap;
+
+  public RotationManager(MatchManager matchManager, File rotationsFile) {
+    this.matchManager = matchManager;
+    this.rotationsFile = rotationsFile;
+
+    if (!rotationsFile.exists()) {
+      try {
+        FileUtils.copyInputStreamToFile(
+            PGM.GLOBAL.get().getResource("rotations.yml"), rotationsFile);
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+    }
+
+    loadRotations();
+    setActiveRotation(getRotationByName("default"));
```

Comment:
I wouldn't do these things in a construtor. Can execute these from outside instead? Also `setActiveRotation` is already called in `loadRotation()`. https://github.com/Electroid/PGM/pull/114/files#diff-485c495f5a184b3547785959ac5276d4R64